notebooks/00-release-28.0.0/03-full-timeline-queries.ipynb

```diff
@@ -0,0 +1,584 @@
+    "    while not done:\n",
+    "        result = datagen.get_json(f\"/status/{job_name}\",'')\n",
+    "        clear_output(wait=True)\n",
+    "        print(json.dumps(result, indent=2))\n",
+    "        if result[\"status\"] == 'COMPLETE':\n",
+    "            done = True\n",
+    "        else:\n",
+    "            time.sleep(1)\n",
+    "\n",
+    "\n",
+    "# monitor ingestion by counting the rows ingested until the expected number of rows have been loaded\n",
+    "def monitor_ingestion( target_table:str, target_rows:int):\n",
+    "    row_count=0\n",
+    "    while row_count<target_rows:\n",
+    "        res = sql_client.sql(f'SELECT count(1) as \"count\" FROM {target_table}')\n",
+    "        clear_output(wait=True)\n",
+    "        print(json.dumps(res, indent=2))\n",
+    "        row_count = res[0]['count']\n",
+    "        time.sleep(1)\n",
+    "        \n",
+    "# suspend the streaming ingestion job and wait for tasks to publish their segments\n",
+    "def stop_streaming_job( target_table: str, reset_offsets: bool = False):\n",
+    "    print(f'Pause streaming ingestion: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/suspend\",\"\", require_ok=False)}]')\n",
+    "    \n",
+    "\n",
+    "    tasks = druid.tasks.tasks(state='running', table=target_table)\n",
+    "    tasks_done = 0\n",
+    "    while tasks_done<len(tasks):\n",
+    "        tasks_done = 0\n",
+    "        clear_output( wait=True)\n",
+    "        print(f'Waiting for running tasks to publish their segments ...')\n",
+    "        for task in tasks:\n",
+    "            status = druid.tasks.task_status(task['id'])\n",
+    "            print(f\"Task [{task['id']}] Status:{status['status']['statusCode']} RunnerStatus:{status['status']['runnerStatusCode']}\")\n",
+    "            if (status['status']['statusCode']!='RUNNING'): \n",
+    "                tasks_done += 1 \n",
+    "        time.sleep(1)\n",
+    "            \n",
+    "    if reset_offsets:\n",
+    "        print(f'Reset offsets for re-runnability: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/reset\",\"\", require_ok=False)}]')\n",
+    "    print(f'Terminate streaming ingestion: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/terminate\",\"\", require_ok=False)}]')\n",
+    "\n",
+    "# Remove table data and metadata from Druid\n",
+    "def drop_table( target_table: str):\n",
+    "    # mark segments as unused \n",
+    "    druid.datasources.drop(target_table)\n",
+    "    # remove segment metadata and data for unused segments\n",
+    "    headers = {'Content-Type': 'application/json'}\n",
+    "    kill_task = {\n",
+    "      \"type\": \"kill\",\n",
+    "      \"dataSource\": target_table,\n",
+    "      \"interval\" : \"2000-09-12/2999-09-13\"\n",
+    "    }\n",
+    "    print(druid.rest.post(f\"/druid/indexer/v1/task\", json.dumps(kill_task),require_ok=False, headers=headers))"
+   ]
+  },
```

Comment:
Does the above need to be in this notebook? Could it be in the API instead?

Reply:
we should add some of these things to the APIs... not doing it now though.
We need a real plan for the druidapi and for the polarisapi. But not for 28.0.0